src/main/java/seedu/address/model/person/Attendance.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a Student's Attendance.
+ */
+public class Attendance {
+    private final LocalDateTime date;
+    private final Boolean hasAttended;
+
+    /**
+     * Represents an Attendance record for a person.
+     * Each Attendance object stores whether the person has attended and the date of attendance.
+     *
+     * @param hasAttended A Boolean indicating if the person has attended.
+     * @param date The date and time of the attendance.
+     * @throws NullPointerException if {@code hasAttended} or {@code date} is null.
+     */
+    public Attendance(Boolean hasAttended, LocalDateTime date) {
```

Comment:
Does `hasAttended` have to be a `Boolean`, instead of the primitive `boolean`? Removes the need for `requireNonNull`.

Reply:
good catch!